### PR TITLE
Add User Agent for Sqs Calls made using Automatic Batching Manager 

### PR DESCRIPTION
**Pull request by joviegas**
## Motivation and Context

- Calls made using High Level library should be identified as `hll/faturename`

## Modifications

- Update override configuration of request to send `hll/abm` in the request

## Testing

- Junits for validate the header of user-agent

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ReceiveSqsMessageHelper.java [hunk 1]
```diff
@@ -72,7 +74,8 @@ public CompletableFuture<ReceiveSqsMessageHelper> asyncReceiveMessage() {
         ReceiveMessageRequest.Builder request =
             ReceiveMessageRequest.builder()
                                  .queueUrl(queueUrl)
-                                 .maxNumberOfMessages(config.maxBatchItems());
+                                 .maxNumberOfMessages(config.maxBatchItems())
+                .overrideConfiguration(o -> o.applyMutation(USER_AGENT_APPLIER));
```
Comment (davidh44) [inline services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ReceiveSqsMessageHelper.java:78]: nit: whitespace alignment

File: services/sqs/src/test/java/software/amazon/awssdk/services/sqs/batchmanager/ReceiveSqsMessageHelperTest.java [hunk 2]
```diff
@@ -291,6 +292,7 @@ public void asyncReceiveMessageArgs() throws Exception {
                                                                      .messageAttributeNames("custom1", "custom2")
                                                                      .visibilityTimeout(9)
                                                                      .waitTimeSeconds(15)
+            .overrideConfiguration(o -> o.applyMutation(SqsMessageDefault.USER_AGENT_APPLIER))
```
Comment (davidh44) [inline services/sqs/src/test/java/software/amazon/awssdk/services/sqs/batchmanager/ReceiveSqsMessageHelperTest.java:295]: nit: whitespace alignment

**Review by davidh44 — APPROVED** [review]

File: services/sqs/src/test/java/software/amazon/awssdk/services/sqs/batchmanager/ReceiveSqsMessageHelperTest.java (hunk 2 above)
Comment (joviegas) [reply services/sqs/src/test/java/software/amazon/awssdk/services/sqs/batchmanager/ReceiveSqsMessageHelperTest.java]: done

File: services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ReceiveSqsMessageHelper.java (hunk 1 above)
Comment (joviegas) [reply services/sqs/src/main/java/software/amazon/awssdk/services/sqs/internal/batchmanager/ReceiveSqsMessageHelper.java]: done

**Pull request merged**